_books/ion-1-1/src/modules/defining_modules.md

```diff
@@ -0,0 +1,242 @@
+
+* An unqualified _macro-name_ is looked up within the exported macros, and if not found, then the active encoding module's macro table.
+  If it maps to a macro, that’s the resolution of the reference.
+  Otherwise, an error is signaled due to an unbound reference.
+* An anonymous local reference  (`__address__`) is resolved by index in the exported macro array.
+  If the address exceeds the array boundary, an error is signaled due to an invalid reference.
+* A qualified reference (`__module__::__name-or-address__`) resolves solely against the referenced module.
+  If the module name does not exist in the visible modules, an error is signaled due to an unbound reference.
+  Otherwise, the name or address is resolved within that module’s exported macro array.
+
+> [!WARNING]
+> An unqualified macro name can change meaning in the middle of a module if you choose to shadow the
+> name of a system macro. To unambiguously refer to the system module, use the qualified reference syntax: `__$ion__::__system-macro-name__`.
+
+
+### `import`
+
+```bnf
+import ::= '(import ' module-binding catalog-key ')'
+catalog-key ::= module-name module-version?
+```
+
+An import binds a lexically scoped module name to a shared module that is identified by a catalog key—a `(name, version)` pair. The `version` of the catalog key is optional—when omitted, the version is implicitly 1.
+
+In Ion 1.0, imports may be substituted with a different version if an exact match is not found.
+In Ion 1.1, however, all imports require an exact match to be found in the reader's catalog;
+if an exact match is not found, the implementation must signal an error.
+
+<!-- TODO: more details here -->
+
+### `module`
+
+The `module` clause defines a new module that is contained in the current module.
+
+```bnf
+inner-module ::= '(module' module-name import* symbol-table? macro-table? ')'
+```
+
+Inner modules automatically have access to modules previously declared in the containing module using `module` or `import`.
+The new module (and its exported symbols and macros) is available to any following `module`, `symbol_table`, and
+`macro_table` clauses in the enclosing container. 
+
+See [inner modules](inner_modules.md) for full explanation.
+
+### `symbol_table`
+
+A module can define a list of exported symbols by copying symbols from other modules and/or declaring new symbols.
+
+```bnf
+symbol-table       ::= '(symbol_table' symbol-table-entry* ')'
+
+symbol-table-entry ::= module-name | symbol-list
+
+symbol-list        ::= '[' ((symbol | string)',')* ']'
+```
+
+The `symbol_table` clause assembles a list of text values for the module to export.
+It takes any number of arguments, each of which may be the name of visible module or a list of symbol-texts.
+The symbol table is a list of symbol-texts by concatenating the symbol tables of named modules and lists of symbol/string values.
+
+Where a module name occurs, its symbol table is appended.
```

Comment:
Should we explicitly mention that `$ion_encoding` is the module name to use to append to the symbol table from the previous context?